### DISCUSSION TITLE
New python-mip release

### DISCUSSION THREAD

**Discussion by AdeelK93** [coin-or/python-mip#373] [General]
Hi! With python 3.12 and highs support - do you think we can release a new version of python-mip to pypi? Thank you!

**Comment by h-g-s** [comment]
Sorry for the delay. Working on a new release in the next few weeks.

**Comment by tkralphs** [comment]
Looks like there's been a new release! https://github.com/coin-or/python-mip/releases/tag/v1.17.1

**Reply by tkralphs** [reply]
Hi @h-g-s, the new release looks great! I noticed you moved the Cbc binaries out of the project, which is definitely a good move. They are now in a binary-only wheel [here](https://pypi.org/project/cbcbox/), but I am wondering if it would be a better solution to just use conda/mamba to install cbc binaries? Obviously, the drawback of that is there is no release yet from the master branch, but I assume we are going to fix that soonish? 

My concern is that that this is now another project that needs long-term maintenance and its releases also don't seem to be tied to the releases of Cbc, making bug reporting difficult. It would be good to have some discussion about these issues to make sure we have a good long-term strategy.

**Reply by rschwarz** [reply]
I don't have the burden of maintenance, but personally, I like the idea of `cbcbox` (also the related `highsbox`) because the solvers are now only a simple `pip install` away. They can also be marked as dependencies for other Python packages, which helps with development.

**Reply by tkralphs** [reply]
Yes, I get the value proposition, but I'm also very sensitive to how these things tend to go in the long-term. Unless someone who is motivated and knows they will have bandwidth into the future steps up to do the maintenance, this will become another pain point. It looks like there's a lot of reinvention of wheels going on inside cbcbox. It has its own build system, which also as to be maintained, while [coinbrew](https://github.com/coin-or/coinbrew) is doing the same job. [CyLP](https://github.com/coin-or/Cbc) already has a much simpler recipe for building Cbc and including binaries in Python wheels. I just want to avoid having yet another complicated build harness that needs to be maintained. 

I guess we can see how things go. If we manage to keep maintaining and making releases of `python-mip`, then someone will hopefully be willing to also look at `cbcbox.`

**Reply by h-g-s** [reply]
> Hi @h-g-s, the new release looks great! I noticed you moved the Cbc binaries out of the project, which is definitely a good move. They are now in a binary-only wheel [here](https://pypi.org/project/cbcbox/), but I am wondering if it would be a better solution to just use conda/mamba to install cbc binaries? Obviously, the drawback of that is there is no release yet from the master branch, but I assume we are going to fix that soonish?
> 
> My concern is that that this is now another project that needs long-term maintenance and its releases also don't seem to be tied to the releases of Cbc, making bug reporting difficult. It would be good to have some discussion about these issues to make sure we have a good long-term strategy.

Hi Ted, 

Yes, we'll have cbc 3.0 soon, doing lots of tests - will be rock solid :)

cbcbox allowed me to experiment quickly with build options - and I was positively surprised with the strong results that AVX2 optimization provided, so I included **both builds** (avx2 optimized and generic) for x86 64.  It would be more convenient in the future if some existing project (i.e. cbc tself) could provide builds with similar features/performance so that we don't bifurcate efforts. As for conda or other alternatives, I'm not knowledgeable of the advantages of these over the approach I took (idea from highsbox) which is distribute wheels from pypi.  Would like to learn more.
